It's usually to be using a debian/ prefix to distinguish Debian tags
from upstream tags. It's also possible to create several revisions
for the same upstream tag, which are encoded as ~pirogue1, ~pirogue2,
etc. in the changelog, but as _pirogue1, _pirogue2, etc. in git tags
(since ~ is a reserved character).

Therefore, drop the debian/ prefix as well as any ~pirogueN suffix
before parsing version information.
--- a/releng/frida_version.py
+++ b/releng/frida_version.py
@@ -4,6 +4,7 @@ import argparse
 from dataclasses import dataclass
 import os
 from pathlib import Path
+import re
 import subprocess
 import sys
 
@@ -45,6 +46,9 @@ def detect(repo: Path) -> FridaVersion:
                                      capture_output=True,
                                      encoding="utf-8").stdout
 
+        description = re.sub(r'^(auto/|)debian/', '', description)
+        description = re.sub(r'_pirogue\d+', '', description)
+
         tokens = description.strip().replace("-", ".").split(".")
         if len(tokens) > 1:
             (raw_major, raw_minor, raw_micro, raw_nano, commit) = tokens
